### PR TITLE
refactor: Remove vestigial gRPC IOX API

### PR DESCRIPTION
**Pull request by alamb**
Per https://github.com/influxdata/influxdb_iox/issues/506, we do not plan to have any gRPC based IOX api, so remove the vestigial code, and rename the other service to reflect its use (testing).

Reviewing this PR commit-by-commit might help as that might make it easier to see what was removed and what was just renamed

### CODE REVIEW

File: generated_types/test.proto [hunk 1]
```diff
@@ -0,0 +1,16 @@
+// This file defines a gRPC service used for testing
```
Comment (alamb) [inline generated_types/test.proto:1]: This was extracted from `test.proto`

File: src/server/rpc/storage.rs [hunk 2]
```diff
@@ -5,7 +5,7 @@
 use std::{collections::HashMap, sync::Arc};
 
 use generated_types::{
-    i_ox_server::{IOx, IOxServer},
+    i_ox_testing_server::{IOxTesting, IOxTestingServer},
```
Comment (e-dard) [inline src/server/rpc/storage.rs:8]: well that's an unfortunate generated name.

**Review by e-dard — APPROVED** [review]
LGTM

FYI - any commit that changes source code shouldn't be a `chore` commit, but rather would be a `refactor`.

File: src/server/rpc/storage.rs (hunk 2 above)
Comment (alamb) [reply src/server/rpc/storage.rs:8]: Indeed

**Comment by alamb** [comment]
> FYI - any commit that changes source code shouldn't be a chore commit, but rather would be a refactor.

I have fixed  on rebase -- thanks for the heads up

**Pull request merged**